### ISSUE TITLE
write score as element

### ISSUE THREAD

**Issue by rettinghaus** [rism-ch/verovio#707]
On "exporting" a MEI file, the internal `<score>` gets lost and a new XML element is created instead (even in *score-based*). To keep IDs (and other attributes), proper reading and writing of `<score>` should be added to `iomei`.

**Comment by rettinghaus** [comment]
Question: What is the advantage in using a `scoreBuffer` instead of a `mdivBuffer`?

**Comment by rettinghaus** [comment]
… and yes, it relates to #659.

**Comment by lpugin** [comment]
I am not sure what to do with this. I cannot how Verovio could render more than one score at the time, so this is why a score buffer is used. Changing this to a mDiv buffer would not solve the issue of not having multiple mDivs, wouldn't it?

**Comment by rettinghaus** [comment]
No, it wouldn't solve the unsolvable "multiple mdivs" problem. And I know that the `<score>` element isn't needed for SVG structure. But it is for import/export/conversion.

Take a simple file like this:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="4.0.0">
    <meiHead xml:id="head01">
        <fileDesc xml:id="head02">
            <titleStmt xml:id="head03">
                <title xml:id="head04" />
            </titleStmt>
        </fileDesc>
    </meiHead>
    <music xml:id="musc-0101">
        <body n="1" xml:id="b-8234" label="the body">
            <mdiv n="1" xml:id="d-7295" type="single">
                <score n="1" xml:id="sc-5223" label="Score">
                    <scoreDef clef.shape="G" clef.line="2" key.mode="minor" key.pname="e" meter.count="4" meter.unit="4" xml:id="sd-7452">
                        <staffGrp xml:id="sg-5135">
                            <staffDef n="1" lines="5" xml:id="st-3453" />
                        </staffGrp>
                    </scoreDef>
                    <section xml:id="p-4754">
                        <measure xml:id="m-3434">
                            <staff n="1" xml:id="s-4564">
                                <layer n="1" xml:id="l-3456">
                                    <note dur="4" oct="4" pname="e" xml:id="n-9384" />
                                </layer>
                            </staff>
                        </measure>
                    </section>
                </score>
            </mdiv>
        </body>
    </music>
</mei>
```

Use Verovio to convert it (from MEI) to MEI you get this: 
```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="http://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="4.0.0">
    <meiHead xml:id="head01">
        <fileDesc xml:id="head02">
            <titleStmt xml:id="head03">
                <title xml:id="head04" />
            </titleStmt>
        </fileDesc>
    </meiHead>
    <music>
        <body>
            <mdiv>
                <score>
                    <scoreDef xml:id="sd-7452" clef.shape="G" clef.line="2" key.mode="minor" key.pname="e" meter.count="4" meter.unit="4">
                        <staffGrp xml:id="sg-5135">
                            <staffDef xml:id="st-3453" n="1" lines="5" />
                        </staffGrp>
                    </scoreDef>
                    <section xml:id="p-4754">
                        <measure xml:id="m-3434">
                            <staff xml:id="s-4564" n="1">
                                <layer xml:id="l-3456" n="1">
                                    <note xml:id="n-9384" dur="4" oct="4" pname="e" />
                                </layer>
                            </staff>
                        </measure>
                    </section>
                </score>
            </mdiv>
        </body>
    </music>
</mei>
```

As you see, all information on `score` and `mdiv` is lost. Although this happens also on `music` and `body`, one could argue that of course Verovio "extracts" one `mdiv` for rendering and other things and puts it back together. But losing the information from `score` isn't understandable, as we have a score representation in Verovio. 

This relates to #659 because I *could* write `@label`, `@type`, and `@n` to `<score>` but it won't find it's way to the resulting MEI.

**Comment by lpugin** [comment]
Yes. I can see that the conversion to the page-based structure creates some confusion and maybe it would be better for Verovio not to do it. Then users would just need to apply an XSLT to convert their MEI files to the Verovio customization externally. This would avoid this type of issues. I will think about it.

**Comment by lpugin** [comment]
One thing that can be done is to add an MDiv class (child of Doc) and an Pages class (child of MDiv). Then attribues in Score should be passed to Pages when transforming to page-based, are reverse when outputting score-based. I expect this to break quite a few things because in many places Page object expect to have a direct Doc parent, but all these should be easy to fix.

Eventually, we can support multiple MDiv (with Score or Pages children), but only one will be active as we do for editorial markup.

Are you willing to do this?

**Comment by rettinghaus** [comment]
This sounds like a good idea, but remember that `<score n="1">` is not identical (or comparable) with `<page n="1">`. Having Mdiv and Page classes makes sense. But at first a somewhat easier approach seems to me to have a vrvVirtualBuffer for holding and manipulate `mdiv` and `score` (maybe `body` as well) with their respective attributes.

**Comment by lpugin** [comment]
It would be `<score n="1">` being comparable to `<pages n="1">`.

I am not sure storing this somewhere else will be easier. If we fix it, we better fix it once for all (at least as much as we can until something is decided for the page-based customisation).

**Comment by lpugin** [comment]
For the record, the customisation currently is:
```xml
<music>
    <body>
        <mdiv>
            <pages>
                <page>
                    <system>
                        <!--... -->
                    </system>
                </page>
            </pages>
        </mdiv>
    </body>
</music>
```

**Comment by lpugin** [comment]
Implemented in b03ac695a1b218412074e0044e73a60cc44bce18

All `<mdiv>` and `<score>` elements are preserved when reading and writing, with `@n` and `@label`. Only a `<mdiv>` with a single `<score>` descendant can be rendered, `mdivXPath` query remain unchanged.

**Comment by rettinghaus** [comment]
looks perfect! Thanks!
